packages/apis/ipfs/src/query/index.ts

```diff
@@ -0,0 +1,46 @@
+import { Http_Query, Http_ResponseType, Http_Body } from "./w3/imported";
+import { 
+  Input_catToString,
+  Input_catFile
+} from "./w3";
+import {decode} from "as-base64"
+
+const EmptyBody: Http_Body = {rawBody:null, stringBody:null, formDataBody: {data: []}}
+
+export function catToString(input: Input_catToString): String {
+  const url = input.ipfsUrl + "/api/v0/cat";
+  const catResponse = Http_Query.get({
+    url: url,
+    request: {
+      headers: [],
+      urlParams: [{key: "arg", value: input.cid}],
+      responseType: Http_ResponseType.TEXT,
+      body: EmptyBody,
+    }
+  });
+
+  if(catResponse == null || catResponse.status != 200) {
+    throw new Error(`Failed to cat file: ${catResponse?.status} ${catResponse?.statusText}`);
+  }
+
+  return catResponse.body;
+}
+
+export function catFile(input: Input_catFile): ArrayBuffer {
+  const url = input.ipfsUrl + "/api/v0/cat";
+  const catResponse = Http_Query.get({
+    url: url,
+    request: {
+      headers: [],
+      urlParams: [{key: "arg", value: input.cid}],
+      responseType: Http_ResponseType.BINARY,
```

Comment:
Http_ResponseType can be made into an option within the `CatFileOptions`. This way users can specify if they'd like the response as BINARY or TEXT

Reply:
In the case of _BINARY_ responseType - the return type of `catFile` function should be `ArrayBuffer` and in the case of _TEXT_ responseType - the return type of `catFile` function should be `String`. 

@dOrgJelli  This means that the actual return type should be `ArrayBuffer | String` and that this also relies on union types #246

Maybe we can open a separate issue for this just not to get lost (I left an in-code comment as additional reminder)